src/Microsoft.XmlSerializer.Generator/tests/Microsoft.XmlSerializer.Generator.Tests.csproj

```diff
@@ -4,21 +4,13 @@
     <DefineConstants>$(DefineConstants);XMLSERIALIZERGENERATORTESTS</DefineConstants>
     <Configurations>netcoreapp-Debug;netcoreapp-Release;uap-Debug;uap-Release</Configurations>
     <CoverageSupported>false</CoverageSupported>
+    <SkipTestsOnPlatform Condition="'$(TargetsUap)' == 'true' OR '$(ArchGroup)' == 'arm' OR '$(ArchGroup)' == 'arm64' OR '$(ArchGroup)' == 'armel'">true</SkipTestsOnPlatform>
   </PropertyGroup>
   <PropertyGroup>
-    <SkipTestsOnPlatform Condition="'$(TargetGroup)' == 'uap' OR '$(ArchGroup)' == 'arm' OR '$(ArchGroup)' == 'arm64' OR '$(ArchGroup)' == 'armel'">true</SkipTestsOnPlatform>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetGroup)' == 'netcoreapp'">
-    <!-- We're building netcoreap, run on the test CLI
-         Reuse the same runtimeconfig that the tests use. -->
-    <GeneratorRuntimeConfig>$(TestAssetsDir)netcoreapp.runtimeconfig.json</GeneratorRuntimeConfig>
-    <GeneratorCliPath>$(TestHostRootPath)</GeneratorCliPath>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetGroup)' != 'netcoreapp'">
-    <!-- We're building some non-netcoreapp target, run on the Tools CLI.
-         Reuse the same runtimeconfig used by MSBuild. -->
-    <GeneratorRuntimeConfig>$(MSBuildToolsPath)\msbuild.runtimeconfig.json</GeneratorRuntimeConfig>
-    <GeneratorCliPath>$(DotNetRoot)</GeneratorCliPath>
+    <!-- Reuse the same runtimeconfig used by MSBuild. -->
```

Comment:
This only works when the build shared-framework and test shared-framework are the same.  I bet that will break very soon when we start building for vNext before we have an SDK that includes it.

Reply:
This should continue to work because of the roll-forward policy: https://github.com/dotnet/corefx/pull/38151/files#diff-6af3f4198209aeccc6388bda6063022cR43

Reply:
It won’t roll forward if it finds exact match and I don’t see DOTNET_MULTILEVEL_LOOKUP.  Full stack build will set this but not a direct build of the project. If you intend to rely on rollforward please add a comment.

Reply:
I get your point here for the NetCoreApp path. I changed it now so that it doesn't rely on roll forward anymore and expects the exact version, the `$(ProductVersion)`. 

As we are already here, this whole assembly looks to me like a big hack and should instead be an msbuild task.

Reply:
I believe it is a tool so that it can be run independently.  @mconnew and his team own it, so please run any large changes through them.  We've had similar conversations in the past.